torchaudio/transforms.py

```diff
@@ -640,6 +641,83 @@ def forward(self, complex_specgrams, overriding_rate=None):
         return F.phase_vocoder(complex_specgrams, rate, self.phase_advance)
 
 
+class Fade(torch.nn.Module):
+    r"""Add a fade in and/or fade out to an waveform.
+
+    Args:
+        fade_in_len (int, optional): Length of fade-in (time frames). (Default: ``0``)
+        fade_out_len (int, optional): Length of fade-out (time frames). (Default: ``0``)
+        fade_shape (str, optional): Shape of fade. Must be one of:
+                                                    "q" for quarter sine,
+                                                    "h" for half sine,
+                                                    "t" for linear,
+                                                    "l" for logarithmic
+                                                    "e" for exponential. (Default: ``"t"``)
+    """
+    def __init__(self, fade_in_len=0, fade_out_len=0, fade_shape="t"):
+        super(Fade, self).__init__()
+        self.fade_in_len = fade_in_len
+        self.fade_out_len = fade_out_len
+        self.fade_shape = fade_shape
+
+    def forward(self, waveform):
+        # type: (Tensor) -> Tensor
+        r"""
+        Args:
+            waveform (torch.Tensor): Tensor of audio of dimension (..., time).
+
+        Returns:
+            torch.Tensor: Tensor of audio of dimension (..., time).
+        """
+        waveform_length = waveform.size()[-1]
+
+        return self._fade_in(waveform_length) * self._fade_out(waveform_length) * waveform
+
+    def _fade_in(self, waveform_length):
+        # type: (int) -> Tensor
+        fade = torch.linspace(0, 1, self.fade_in_len)
+        ones = torch.ones(waveform_length - self.fade_in_len)
+
+        if self.fade_shape == "t":
+            fade = fade
+
+        if self.fade_shape == "e":
+            fade = torch.pow(2, (fade - 1)) * fade
+
+        if self.fade_shape == "l":
+            fade = torch.log10(.1 + fade) + 1
+
+        if self.fade_shape == "q":
+            fade = torch.sin(fade * math.pi / 2)
+
+        if self.fade_shape == "h":
+            fade = torch.sin(fade * math.pi - math.pi / 2) / 2 + 0.5
+
+        return torch.cat((fade, ones)).clamp_(0, 1)
+
+    def _fade_out(self, waveform_length):
+        # type: (int) -> Tensor
+        fade = torch.linspace(0, 1, self.fade_out_len)
+        ones = torch.ones(waveform_length - self.fade_out_len)
+
+        if self.fade_shape == "t":
+            fade = - fade + 1
+
+        if self.fade_shape == "e":
+            fade = torch.pow(2, - fade) * (1 - fade)
+
+        if self.fade_shape == "l":
+            fade = torch.log10(1.1 - fade) + 1
+
+        if self.fade_shape == "q":
+            fade = torch.sin(fade * math.pi / 2 + math.pi / 2)
+
+        if self.fade_shape == "h":
+            fade = torch.sin(fade * math.pi + math.pi / 2) / 2 + 0.5
+
+        return torch.cat((ones, fade)).clamp_(0, 1)
```

Comment:
Is `clamp` necessary? Should we let the user decide what to do with this? Which step would you expect would make the signal go outside of `[0,1]`?

Reply:
To create fade-in and fade-out is calculated a percentage by which the audio or signal will be multiplied, this way simulating the fade. So, to guarantee that it will be between [0, 1] I added `clamp_` to enforce this.